source/extensions/http/cache/file_system_http_cache/file_system_http_cache.cc

```diff
@@ -388,6 +404,108 @@ void FileSystemHttpCache::trackFileRemoved(uint64_t file_size) {
   stats_.size_bytes_.set(size_bytes_);
 }
 
+bool FileSystemHttpCache::needsEviction() const {
+  if (config().has_max_cache_size_bytes() &&
+      size_bytes_ > config().max_cache_size_bytes().value()) {
+    return true;
+  }
+  if (config().has_max_cache_entry_count() &&
+      size_count_ > config().max_cache_entry_count().value()) {
+    return true;
+  }
+  return false;
+}
+
+bool FileSystemHttpCache::needsEviction(uint64_t& size_to_evict_out,
```

Comment:
Can we merge the two implementations for needsEviction and have it return the two eviction-goals as a pair?

We can then just check the value of those for non-zero at the call-site.

Reply:
Done. (A struct rather than a pair to make it nicer to work with.)